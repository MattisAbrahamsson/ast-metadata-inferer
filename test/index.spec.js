import astNodeTypeChecker from '../'

jest.setTimeout(20000)

describe('AstNodeTypeChecker', () => {
  it('should return basic output', async () => {
    const [record] = await astNodeTypeChecker();
    expect(record).toHaveProperty('apiType');
    expect(record).toHaveProperty('type');
    expect(record).toHaveProperty('protoChain');
    expect(record).toHaveProperty('protoChainId');
    expect(record).toHaveProperty('astNodeType');
    expect(record).toHaveProperty('isStatic');
    expect(record).toMatchSnapshot()
  })
})
