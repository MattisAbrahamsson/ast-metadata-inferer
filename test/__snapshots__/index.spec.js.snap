// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AstNodeTypeChecker should return basic output 1`] = `
Object {
  "apiType": "js-api",
  "astNodeType": Array [
    "MemberExpression",
  ],
  "isStatic": false,
  "protoChain": Array [
    "ANGLE_instanced_arrays",
  ],
  "protoChainId": "ANGLE_instanced_arrays",
  "type": "js-api",
}
`;
